services/movie_session.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+from typing import Optional
+
+from db.models import MovieSession
+from django.db.models import QuerySet
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(session_date: Optional[datetime] = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    queryset = MovieSession.objects.get(id=movie_session_id)
+    return queryset
+
+
+def update_movie_session(session_id: int,
+                         show_time: Optional[datetime] = None,
+                         movie_id: Optional[int] = None,
+                         cinema_hall_id: Optional[int] = None) -> None:
+    queryset = MovieSession.objects.filter(id=session_id)
+    if show_time:
+        queryset.update(show_time=show_time)
+
+    if movie_id:
+        queryset.update(movie_id=movie_id)
+
+    if cinema_hall_id:
+        queryset.update(cinema_hall_id=cinema_hall_id)
```

Comment:
YOu can do it in 2 queries. No need to use .update because each .update do 1 query